src/search/utils/collections.h

```diff
@@ -151,6 +152,98 @@ int estimate_unordered_map_bytes(int num_entries) {
     // See comments for _estimate_hash_table_bytes.
     return _estimate_hash_table_bytes<std::unordered_map<Key, Value>>(num_entries);
 }
+
+    /**
+     * NamedVector is a vector-like collection with optional names to be associated with each element.
+     * It is optimized to have minimal overhead when there are no names.
+     * Any name which is not specified is assumed to be an empty string.
+     */
+template<typename T>
+class NamedVector {
```

Comment:
I think this new class should be moved to its own file. I think it fits best under algorithms/named_vector.h, given the other files in that directory, even though this is more a data structure than an algorithm.

Reply:
I think I'll wait for a second opinion on this before moving this class. @FlorianPommerening what do you think?

Reply:
I'm fine with that. As Jendrik said, the name "algorithms" is not that good of a fit, but this is already the case for the things that are there. (It should probably be "algorithms and data structures".)